lib/api/completeMultipartUpload.js

```diff
@@ -221,18 +222,25 @@ function completeMultipartUpload(authInfo, request, log, callback) {
                 calculatedSize = completeObjData.contentLength;
                 return next(null, destBucket, objMD, mpuBucket,
                     storedMetadata, completeObjData.eTag, calculatedSize,
-                    null, mpuOverviewKey, null, null, completeObjData);
+                    null, [mpuOverviewKey], null, completeObjData);
             }
-            const storedPartsAsObjects = storedParts.map(item => ({
-                // In order to delete the part listing in the shadow
-                // bucket, need the full key
-                key: item.key,
-                ETag: `"${item.value.ETag}"`,
-                size: item.value.Size,
+            let storedPartsAsObjects = [];
+            const keysToDelete = [];
+            for (let i = 0; i < storedParts.length; i++) {
```

Comment:
Yes forEach is probably more appropriate.
Interesting article I found: http://thejsguy.com/2016/07/30/javascript-for-loop-vs-array-foreach.html